NitroxClient/GameLogic/Spawning/Metadata/EntityMetadataManager.cs

```diff
@@ -55,11 +62,27 @@ public Optional<IEntityMetadataProcessor> FromMetaData(EntityMetadata metadata)
 
     public void ApplyMetadata(GameObject gameObject, EntityMetadata metadata)
     {
+        // In case a metadata update was received while this entity was in the spawn queue
+        if (gameObject.TryGetNitroxId(out NitroxId objectId) &&
+            newerMetadataById.TryGetValue(objectId, out EntityMetadata newMetadata))
+        {
+            metadata = newMetadata;
```

Comment:
The way this works is: (in a chronological order)
- You receive X entities to spawn
- While you are spawning those entities (over multiple frames so other events happen and packet are processed) you receive a metadata update packet and you store it as you can't apply it yet (you haven't spawned the target entity)
- When spawning an entity, you check if it has some newer metadata and if there is one, you only need to apply the newer one and not the one you had inside of the entity object because it is now obsolete.